lib/jit_preloader/active_record/base.rb

```diff
@@ -41,17 +41,39 @@ def has_many_aggregate(assoc, name, aggregate, field, default: 0)
           association_scope = klass
           association_scope = association_scope.instance_exec(&reflection.scope).reorder(nil) if reflection.scope
 
-          conditions[reflection.foreign_key] = primary_ids
+          if reflection.options.keys.include?(:through)
+            source_class = self.class
 
-          if reflection.type.present?
-            conditions[reflection.type] = self.class.name
+            through = reflection.options[:through]
+            reflections = []
+            while through do
+              through_reflection = source_class.reflect_on_association(through)
+              reflections << through_reflection
+              through = through_reflection.options[:through]
+            end
+
+            conditions["#{source_class.table_name}.#{source_class.primary_key}"] = primary_ids
+
+            join_for_group_by = "INNER JOIN #{source_class.table_name} on #{reflections.last.table_name}.#{reflections.last.foreign_key} = #{source_class.table_name}.#{source_class.primary_key}"
```

Comment:
This code here is fairly dense, and hard to follow. 

We are also rebuilding some internal Rails logic to construct the joins and I am not 100% confident that it covers all of the edge cases. (Though your specs have been a great help!)

I think there is a way we can leverage the Rails internals to do this work for us, and at the same time, not worry about having to have a large branch on the `through`

Reply:
@doliveirakn Yes, this looks great! I like that there isn't a separate if branch for the through relationships, looks much cleaner now. I really like the idea of ActiveRecord constructing the query and not doing it manually, I couldn't quite figure out how to do that before but I'm glad that you looked through and did this.  

I went ahead and modified this PR. The only thing I changed in your code example was the naming of first_association to aggregate_association which I thought was a little easier to follow but really wasn't a big deal if you want me to change that back I can.